modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotValidator.scala

```diff
@@ -132,29 +133,60 @@ object CurrencySnapshotValidator {
         }
       })
 
-      val recreateFn = (trigger: ConsensusTrigger) =>
+      def recreateFn(trigger: ConsensusTrigger) =
         mkEvents.flatMap { events =>
-          currencySnapshotCreator
-            .createProposalArtifact(lastArtifact.ordinal, lastArtifact, lastContext, trigger, events, rewards, facilitators)
+          def usingKryo =
+            currencySnapshotCreator
+              .createProposalArtifact(
+                lastArtifact.ordinal,
+                lastArtifact,
+                lastContext,
+                Hasher.forJson,
```

Comment:
Shouldn't be Hasher.forKryo?

Reply:
Yeah, it will work with both ways: Json with Kryo fallback and Kryo with Json fallback but semantically it is a mistake and should be reversed.

Reply:
yep, you're right @IPadawans